gnucash/register/ledger-core/split-register.c

```diff
@@ -2691,6 +2691,14 @@ split_register_pref_changed (gpointer prefs, gchar *pref, gpointer user_data)
     {
         info->separator_changed = TRUE;
     }
+    else if (g_str_has_suffix(pref, GNC_PREF_SHOW_LEAF_ACCT_NAMES))
+    {
+        reg->show_leaf_accounts = !reg->show_leaf_accounts;
```

Comment:
This is a potentially dangerous construct. If the setting-changed signal fails to trigger at some point, or triggers twice while the option is only changed once show_leaf_accounts would go out of sync with the real option.
Considering the option change doesn't happen that often, I would prefer to just use gnc_prefs_get_bool here. That way you're sure to always get the current value of the preference.

Other than that, nice work. Thanks!

Reply:
I had it that way to start with but thought this way was simpler but I can see your point so expect a new PR to change it shortly.